pkg/storage/kubernetes/ipam.go

```diff
@@ -542,10 +542,10 @@ func IPManagementKubernetesUpdate(ctx context.Context, mode int, ipam *Kubernete
 				}
 
 			case whereaboutstypes.Deallocate:
-				updatedreservelist, ipforoverlappingrangeupdate, err = allocate.DeallocateIP(reservelist, containerID)
-				if err != nil {
-					logging.Errorf("Error deallocating IP: %v", err)
-					return newips, err
+				updatedreservelist, ipforoverlappingrangeupdate = allocate.DeallocateIP(reservelist, containerID)
+				if updatedreservelist == nil {
+					// Do not fail if allocation is not present.
```

Comment:
Cool, I'm following. I like that you logged the condition in `allocate.DeallocateIP()` -- thanks -- and can silently pass here otherwise. Nice one.

Reply:
that's exactly what I don't like about it lol. 

I'm ok with it though, since we do not unit test that function.